Lib.AspNetCore.ServerTiming/ServerTimingMiddlewareExtensions.cs

```diff
@@ -48,14 +49,34 @@ public static IApplicationBuilder UseServerTiming(this IApplicationBuilder app,
         /// <param name="app">The <see cref="IApplicationBuilder"/> passed to Configure method.</param>
         /// <param name="timingAllowOrigins">The origins that are allowed to see values from timing APIs.</param>
         /// <returns>The original app parameter</returns>
+        [Obsolete("Use a version of this method that takes a lambda function to intialise options")]
         public static IApplicationBuilder UseServerTiming(this IApplicationBuilder app, params string[] timingAllowOrigins)
         {
             if (app is null)
             {
                 throw new ArgumentNullException(nameof(app));
             }
 
-            return app.UseMiddleware<ServerTimingMiddleware>(new TimingAllowOriginHeaderValue(timingAllowOrigins));
+            return UseServerTiming(app,options => options.AllowedOrigins.AddRange(timingAllowOrigins));
+        }
+
+
+        /// <summary>
+        /// Adds a <see cref="ServerTimingMiddleware"/> to application pipeline.
+        /// </summary>
+        /// <param name="app">The <see cref="IApplicationBuilder"/> passed to Configure method.</param>
+        /// <param name="setOptionsCallback">A lambda to set the configuration options for server timing</param>        
+        /// <returns>The original app parameter</returns>
+        public static IApplicationBuilder UseServerTiming(this IApplicationBuilder app, Action<ServerTimingOptions> setOptionsCallback)
```

Comment:
I'm realy wondering if there is a value in adding this one. using callback is usually tied together with implementing `IOptions` pattern for the middleware, which is not being done (and I'm not sure if it makes sense).

With this implementation it doesn't make difference from building ServerTimingOptions during startup and passing it directly.